quipucords/utils/misc.py

```diff
@@ -34,3 +36,16 @@ def sanitize_for_utf8_compatibility(value):
     if isinstance(value, tuple):
         return tuple(sanitize_for_utf8_compatibility(value) for value in value)
     return value
+
+
+def file_exists(file_path: str | None) -> bool:
```

Comment:
The name suggests it's a wrapper around `path.exists()`, but it checks more than just that.

What about `file_readable` (I guess non-existing file is not readable by definition)? Or `file_accessible`? Or `valid_cache_file` (that in turn implies more than this function can in fact deliver)?

Reply:
Good point on the name. This helper function started simpler, thus the simple name, and grew in complexity as I worked through the issues. I have renamed this function to `is_valid_cache_file`.